zrml/prediction-markets/src/lib.rs

```diff
@@ -3315,7 +3405,21 @@ mod pallet {
             MarketIdsPerBlock::remove(block_number);
 
             let mut time_frame_ids_len = 0u32;
-            for time_frame in last_time_frame.saturating_add(1)..=current_time_frame {
+            let start = last_time_frame.saturating_add(1);
+            let end = current_time_frame;
+            let diff = end.saturating_sub(start);
+            let start = if diff > MAX_RECOVERY_TIME_FRAMES {
+                log::warn!(target: LOG_TARGET, "Could not recover all time frames since the last time frame {:?}.", last_time_frame);
+                let limit_time_frame = end.saturating_sub(MAX_RECOVERY_TIME_FRAMES);
+                Self::deposit_event(Event::RecoveryLimitReached {
+                    last_time_frame,
+                    limit_time_frame,
+                });
+                limit_time_frame
+            } else {
+                start
+            };
```

Comment:
This seems very confusing.

```suggestion
            let end = current_time_frame;
            let start = last_time_frame.max(end.saturating_sub(MAX_RECOVER_TIME_FRAMES));
```

You can then emit the event based on the diff between `start` and `end`. I question the point of the `log::warn!` call. Has anyone _ever_ looked at these logs? Do we use them? Are we notified when a warning is logged?

Reply:
I mean logs are a good way to understand the internal logic if it runs on the nodes. It's generally useful to have the logs. I think @sea212 raised the requirement to emit events and logs for this task in the related issue. So assume the chain stops producing blocks. We want to quickly figure out the problem. In that case, logs help us to understand unexpected behaviour and the general flow of the program.

Reply:
last_time_frame is missing the addition by one. I think I just left it as it was before. Why did we have this addition in the first place?

Reply:
> You can then emit the event based on the diff between start and end.

At what condition would you emit the event?